### PR TITLE
Say.js

### PR DESCRIPTION
**Pull request by etcroot**
Submission to add say command.

### CODE REVIEW

**Comment by kyranet** [comment]
This needs linting, CI is most likely red.

**Comment by bdistin** [comment]
Should use msg.channel.send, because of deletable commands option

**Comment by etcroot** [comment]
Adding it now & re-submitting it

**Comment by tech6hutch** [comment]
Why permission level 4?

**Comment by etcroot** [comment]
> Why permission level 4?

Because a say command could be abused by people who isn't a moderator+

**Comment by tech6hutch** [comment]
Is that also how you're getting around the @everyone issue?

**Comment by etcroot** [comment]
> Is that also how you're getting around the @everyone issue?

Indeed. I believe a say command should have higher perms for the abuse risks.

File: commands/Fun/say.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    super(...args, {
+      description: "Make the bot say something",
+      usage: "<message:string>",
+      permissionLevel: 4,
+      aliases: ["echo"]
+    });
+  }
+
+  async run(msg, [args]) {
+    msg.delete().catch(O_o=>{});
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: I love the `O_o => {}`, but maybe change it to ` () => {}`

File: commands/Fun/say.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      description: "Make the bot say something",
+      usage: "<message:string>",
+      permissionLevel: 4,
+      aliases: ["echo"]
+    });
+  }
+
+  async run(msg, [args]) {
+    msg.delete().catch(O_o=>{}); 
+    return msg.channel.send(args);
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: Please use the disableEveryone option in send. That will prevent issues and @ everyone spam

File: commands/Fun/say.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const { Command } = require("klasa");
+
+class Say extends Command {
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: This doesn't really respect the formst we use in the other pieces. Please use `module.exports = class extends Command`

File: commands/Fun/say.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const { Command } = require("klasa");
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: Please add the copyright header above this line. You can find it in any piece from this repo.

Comment (etcroot) [reply commands/Fun/say.js]: Added it

File: commands/Fun/say.js (hunk 2 above)
Comment (etcroot) [reply commands/Fun/say.js]: Done

File: commands/Fun/say.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2017-2018 dirigeants. All rights reserved. MIT license.
+const { Command } = require("klasa");
+
+module.exports = class extends Command {
+  constructor(...args) {
+    super(...args, {
+      description: "Make the bot say something",
+      name: "say",
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: Why?

File: commands/Fun/say.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      usage: "<message:string>",
+      disableEveryone: true,
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: Message send option not command option :lul:

File: commands/Fun/say.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      permissionLevel: 4,
+      aliases: ["echo"]
+    });
+  }
+
+  async run(msg, [args]) {
+    msg.delete().catch() => {}
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: This is not what I meant... This isn't even valid js

Comment (etcroot) [reply commands/Fun/say.js]: You told me to replace (O_o=>{} to () => {}

Comment (kyranet) [reply commands/Fun/say.js]: > I love the `O_o => {}`, but maybe change it to ` () => {}`

Attention to detail. And he meant the contents of the .catch

Comment (etcroot) [reply commands/Fun/say.js]: > > I love the `O_o => {}`, but maybe change it to ` () => {}`
> 
> Attention to detail. And he meant the contents of the .catch

Yeah, i just fixed & pushed it.

**Comment by etcroot** [comment]
Fixed linting

**Comment by etcroot** [comment]
@bdistin Fixed linting errors now on latest commit.

File: commands/Fun/say.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+const { Command } = require('klasa');
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, {
+			description: 'Make the bot say something',
+			name: 'say',
+			usage: '<message:string>',
+			disableEveryone: true,
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: Message send option NOT COMMAND option /tableflip

File: commands/Fun/say.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+			permissionLevel: 4,
+			aliases: ['echo'],
+			usageDelim: ' '
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: Remove this

File: commands/Fun/say.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2017-2018 dirigeants. All rights reserved. MIT license.
+const { Command } = require('klasa');
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, {
+			description: 'Make the bot say something',
+			name: 'say',
```
Comment (KingDGrizzle) [inline commands/Fun/say.js]: And ditto, remove this, not needed (name is set from file name)

File: commands/Fun/say.js (hunk 8 above)
Comment (etcroot) [reply commands/Fun/say.js]: I don't know how to add a message send option :l

Comment (KingDGrizzle) [reply commands/Fun/say.js]: `message.channel.send(content, { disableEveryone: true })`

Comment (etcroot) [reply commands/Fun/say.js]: Resolved

File: commands/Fun/say.js (hunk 9 above)
Comment (KingDGrizzle) [reply commands/Fun/say.js]: ^

Comment (etcroot) [reply commands/Fun/say.js]: Removed permission level.

Comment (KingDGrizzle) [reply commands/Fun/say.js]: I pointed at the usageDelim not the permissionLevel :notlikethis:

Comment (etcroot) [reply commands/Fun/say.js]: Fixed

**Comment by etcroot** [comment]
All requests fixed.

**Comment by etcroot** [comment]
Added dice command into this PR too.

**Comment by YorkAARGH** [comment]
@etcroot please have one command per PR.

File: commands/Fun/dice.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+	constructor(...args) {
+		super(...args, {
+			aliases: ['roll'],
+			description: 'Roll the dice.'
+		});
+	}
+
+	run(msg) {
+		// eslint-disable-next-line no-mixed-operators
+		const roll = Math.floor(Math.random() * 6 + 1);
```
Comment (kyranet) [inline commands/Fun/dice.js]:
```suggestion
		const roll = Math.floor((Math.random() * 6) + 1);
```

File: commands/Fun/say.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+			description: 'Make the bot say something',
+			usage: '<message:string>',
+			permissionLevel: 4,
+			aliases: ['echo']
+		});
+	}
+
+	async run(msg, [args]) {
+		// eslint-disable-next-line no-empty-function
+		msg.delete().catch(() => {});
```
Comment (kyranet) [inline commands/Fun/say.js]:
```suggestion
		msg.delete().catch(() => null);
```

**Comment by kyranet** [comment]
Bump. Please check the two requested changes (before I can merge).

**Comment by kyranet** [comment]
Closing due to age of requested changes.

**Pull request closed without merging**